graphics/src/MeshManager.cc

```diff
@@ -822,6 +822,87 @@ void MeshManager::CreateCamera(const std::string &_name, float _scale)
   mesh->RecalculateNormals();
 }
 
+void MeshManager::CreateEllipsoid(const std::string &_name,
+                                  const ignition::math::Vector3d _radii,
+                                  const unsigned int _rings,
+                                  const unsigned int _segments)
+{
+  if (this->HasMesh(_name))
+  {
+    return;
+  }
+
+  Mesh *mesh = new Mesh();
+  mesh->SetName(_name);
+  this->dataPtr->meshes.insert(std::make_pair(_name, mesh));
+
+  ignition::math::Vector3d vert, norm;
+  SubMesh subMesh;
+
+  double umin = -M_PI / 2.0;
+  double umax = M_PI / 2.0;
+  double vmin = 0.0;
+  double vmax = 2.0 * M_PI;
+
+  unsigned int i, j;
+  float theta, phi;
+  float d_phi = (umax - umin) / (_rings - 1.0);
+  float d_theta = (vmax - vmin) / (_segments - 1.0);
+
+  float c_theta, s_theta, c_phi, s_phi;
```

Comment:
I don't think these have to be declared outside the loop since they aren't used outside the loop